engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -183,6 +184,13 @@ object Main {
       .longOpt(LANGUAGE_SERVER_PROFILING_EVENTS_LOG_PATH)
       .desc("The path to the runtime events log file.")
       .build()
+    val lsEdition = CliOption.builder
+      .hasArg(true)
+      .numberOfArgs(1)
+      .argName("edition")
+      .longOpt(LANGUAGE_SERVER_EDITION)
+      .desc("Override edition specified in the project config.")
```

Comment:
What is the exact use-case for this option? Can we have some documentation explain how people are supposed to use it and why?

Reply:
The edition specified in the project config can be unavailable which leads to a failure when opening a project. This option allows to mitigate the failure by supplying the fallback version.

Reply:
However opening a project with different edition may lead to "failures when opening a project" - different editions have different language features and different standard libraries APIs.

Reply:
These options are passed to the language server when you selects the project in IDE. They are not user-facing

Reply:
> However opening a project with different edition may lead to "failures when opening a project" - different editions have different language features and different standard libraries APIs.

I would say this is a valid point. Can we make sure that IDE is aware of this override and can report this to the user? Or at least that we log this in our logs (I didn't see anything in this PR)?Otherwise I'm afraid one of us will be chasing this ghost, being unable to reproduce.

Reply:
>  IDE is aware of this override and can report this to the user?

Essential! IDE has to tell the user that wrong edition is used to open the project. E.g. the message needs to be shown.

Ideally it would be a question: _"go on and open with Engine XYZ or give up?"_ - before I can review the API (well CLI) changes, I need to understand the workflow. Can we start from describing the UX and only then do API changes?

CCing @jdunkerley